src/simplay-jupyter/src/accurateSlider.ts

```diff
@@ -148,16 +148,13 @@ export class AccurateSlider {
     this.slider.appendChild(this.leftSegment.segment);
     this.slider.appendChild(this.rightSegment.segment);
 
-    document.onload = () => {
-      setTimeout(() => {
-        this.updateVisual();
-      }, 50);
-    };
+    new ResizeObserver(() => this.updateVisual()).observe(this.slider);
     this.registerEventHandlers();
   }
 
   private createRoot(): HTMLDivElement {
     const root = document.createElement('div');
+    root.classList.add('simplay-accurate-slider');
```

Comment:
This line causes a high degree of coupling. Please parameterize, or append to the slider div directly.